### ISSUE TITLE
Heap corruption in MSVC build.

### ISSUE THREAD

**Issue by lazyhamster** [dscharrer/innoextract#99]
I've tried to build current master branch in Visual Studio 2017 but ran into problem. Newly built program crashes instantly. Looks like heap becomes corrupted.

Message from debugger:

```
HEAP[innoextract.exe]: Invalid address specified to RtlValidateHeap( 014A0000, 00F0ADE0 )
innoextract.exe has triggered a breakpoint.

Debug Assertion Failed!

Program: C:\Temp\cmake-build\Debug\innoextract.exe
File: minkernel\crts\ucrt\src\appcrt\heap\debug_heap.cpp
Line: 904

Expression: _CrtIsValidHeapPointer(block)
```

After some code commenting I've found offending line in **main** function:

`boost::filesystem::path::imbue(std::locale(std::locale(), &util::codecvt));`

Commenting it or replacing with something simpler like

`boost::filesystem::path::imbue(std::locale());`

fixes the problem and program runs properly

**Comment by dscharrer** [comment]
Probably related to issue #76. I plan to replace `boost::filesystem` for the next release so this code will no longer be used.

**Comment by ofry** [comment]
@dscharrer After googling, I found https://stackoverflow.com/questions/32272959/boostfilesystempathimbue-failed-with-default-stdlocale

**Comment by ofry** [comment]
@dscharrer Is it helpful?

**Comment by dscharrer** [comment]
This should be fixed with version 1.9.
